network/p2p/peerManager.go

```diff
@@ -84,8 +84,10 @@ func PeerManagerFactory(peerManagerOptions []Option, connectorOptions ...Connect
 
 // Ready kicks off the ambient periodic connection updates.
 func (pm *PeerManager) Ready() <-chan struct{} {
+	pm.unit.Launch(pm.updateLoop)
+
 	// makes sure that peer update request is invoked once before returning
-	pm.RequestPeerUpdate()
+	pm.peerRequestQ <- struct{}{}
```

Comment:
Previously we were launching the loop *after* invoking peer update request for the first time.

However, if we look at the implementation of `RequestPeerUpdate` below, we see that this first peer update request is *always* dropped immediately.

We need to launch the update loop first, so that by the time the peer update request is triggered, there is actually someone listening to the channel. In fact, calling `unit.Launch(pm.updateLoop)` is not sufficient, because there is still a race condition where the peer update could be attempted before the `updateLoop` goroutine has entered the first loop iteration.

So, actually we need to make this first peer update request blocking, to ensure that it is not dropped.